chroma-manager/tests/integration/core/api_testcase_with_test_reset.py

```diff
@@ -891,160 +886,115 @@ def linux_devices_exist(cls):
-                        zfs_device = TestBlockDevice('zfs', zpool_dataset)
-
-                        self.execute_commands(zfs_device.destroy_commands,
-                                              first_test_server['fqdn'],
-                                              'destroy zfs device %s' % zfs_device)
-
-        if action & self.CZP_CREATEZPOOLS:
-            for lustre_device in config['lustre_devices']:
-                if lustre_device['backend_filesystem'] == 'zfs':
-                    zfs_device = TestBlockDevice('zfs', first_test_server['zpool_device_paths'][lustre_device['path_index']])
-
-                    self.execute_commands(zfs_device.prepare_device_commands,
-                                          first_test_server['fqdn'],
-                                          'create zfs device %s' % zfs_device)
-
-        if action & self.CZP_EXPORTPOOLS:
-            for zfs_device in imported_zpools:
-                self.execute_commands(zfs_device.release_commands,
-                                      first_test_server['fqdn'],
-                                      'export zfs device %s' % zfs_device)
+        zfs_devices = [
+            TestBlockDevice('zfs', x) for x in zfs_device_paths
+        ]
 
-        for server in test_servers:
-            for lustre_device in config['lustre_devices']:
-                if lustre_device['backend_filesystem'] == 'zfs':
-                    zfs_device = TestBlockDevice('zfs', first_test_server['zpool_device_paths'][lustre_device['path_index']])
+        [
+            self.execute_simultaneous_commands(
+                x.clear_device_commands([x.device_path]),
+                fqdns,
+                'destroy zpool %s' % x,
+                expected_return_code=None) for x in zfs_devices
+        ]
 
-                    self.execute_commands(zfs_device.release_commands,
-                                          server['fqdn'],
-                                          'export zfs device %s' % zfs_device,
-                                          expected_return_code=None)
+        def wipe(x):
+            return 'wipefs -a {0} && parted {0} mklabel gpt'.format(x)
 
-    def cleanup_linux_devices(self, test_servers):
-        """
-        Destroy any partitions on the block device.
+        self.execute_commands([wipe(x) for x in zfs_device_paths],
+                              server0['fqdn'], 'wiping disks')
 
-        Very linux specific code.
-        :return:
-        """
-        if (self.simulator is not None) or (self.linux_devices_exist() is False):
+        self.execute_simultaneous_commands(['partprobe', 'udevadm settle'], fqdns, 'sync partitions')
+
+        [
+            self.remote_operations.reset_server(
+                x) for x in fqdns
+        ]
+
+        [
+            self.remote_operations.await_server_boot(
+                x) for x in fqdns
+        ]
+
+    def cleanup_linux_devices(self, test_servers):
```

Comment:
at some point it would be nice if these were shared between efs & ssi so we use consistent cleanup methods

Reply:
Agreed.